### DISCUSSION TITLE
Unable to send data to MindSphere: Unexpected token '='

### DISCUSSION THREAD

**Discussion by Mbonss** [mindsphere/node-red-contrib-mindconnect#155] [Q&A] [answered]
After configuring the MindConnect node with the manual, the node confirms the settings by showing the status information.

![image](https://user-images.githubusercontent.com/91943633/159006729-0d83ea09-d9d5-494f-8a21-d6e6d1a8fdba.png)

SHARED_SECRET Token is retreived from MindSphere, Asset is showing "Onboarded" status.

When sending data to MS using the timeseries template I get the following error:

![image](https://user-images.githubusercontent.com/91943633/159006774-ade6e370-10fb-459c-a94c-59c3e9582d5f.png)

I am using Node-Red v2.1.3, @mindconnect/node-red-contrib-mindconnect version 3.12.0
running on IOT2050 Industrial OS v2.2

So far I am not able to find the root of this error. 
Is this expected behaviour?

**Comment by sn0wcat** [comment]
Can you post the content of your function node, i think you have a typo in there.

**Reply by Mbonss** [reply]
I am just sending 1 variable at the moment, for simplicity.

![image](https://user-images.githubusercontent.com/91943633/159023502-5725ab0f-5702-4bd3-8601-4283cd7422c0.png)

(sorry that it's a snippet, the ` sign ruins it)

**Reply by sn0wcat** [reply]
just copy it here, i will parse it out, please copy the full text of the function. :)

**Reply by Mbonss** [reply]
// take a look at the flow examples at https://playground.mindconnect.rocks
//
// - MindSphere expects all values to be encoded as strings
// - Integers must not have any decimal places
// - It is a good practice to use javascript literal notation
// 
// * Examples: 
// 
//   - Random value between 0 and 100: `${Math.random()*100}` 
//   - Variable (HUMIDITY) from input node : `${msg.payload.HUMIDITY}` 
//   - Variable (SPEED) from input node with 0 as default: `${msg.payload.SPEED || 0}`

const values = [
  {
    "dataPointId": "DP-sinr",
    "qualityCode": "0",
    "value": `${Math.random()*100}`
  }
];
msg._time = new Date();
msg.payload=values;
return msg;

**Comment by sn0wcat** [comment]
can you go to your mindsphere agent diagnostic application on your mindsphere dashboard and switch the agent diagnostic on? I don't see any problem in your function node...
e.g. click on the steoscope icon 
![image](https://user-images.githubusercontent.com/1426617/159028454-1505568b-5df5-46ed-a82d-f38fdfc7cfa0.png)

**Reply by Mbonss** [reply]
Activated the Agent diagnostics in MS, no monitoring logs are displayed..

**Comment by sn0wcat** [comment]
@Mbonss also , how old is this agent? the agent management service of mindsphere had a major outage few days ago, maybe the configuration settings are broken for some reason, You could offboard the agent in the mindsphere, delete the local settings by clicking on the trash can button, and then onboard the agent again.

**Reply by Mbonss** [reply]
Today was the first time I connect the asset to MS. I have tried off and onboarding several times after that. Just tried deleting and recreating the asset : same result.

**Comment by sn0wcat** [comment, marked as answer]
I could reproduce it. The problem only occurs if you install the mindconnect node while building the docker image.

```docker
FROM nodered/node-red:2.1.3
RUN  npm install @mindconnect/node-red-contrib-mindconnect
```
This will show the following error, and the error you are seeing is just a follow up error because the validation function can't find the AJV  library.

```log
npm WARN ajv-keywords@5.1.0 requires a peer of ajv@^8.8.2 but none is installed. You must install peer dependencies yourself.
```

For some reason npm has a problem to handle dependencies in the version 2.1.3 of the image when installing the library via NPM command while building docker container.

This doesn't happen if you use e.g. the latest version of docker image.

```docker
FROM nodered/node-red:latest
RUN  npm install @mindconnect/node-red-contrib-mindconnect
```
It also doesn't happen if you **just install the library via manage palette** instead of building it via docker container. 

A workaround if you absolutely must use that image is to switch off the validation in the node, but I suggest just using the newer base image for your docker container. 

![image](https://user-images.githubusercontent.com/1426617/159064473-42200445-2983-49f0-8800-4cb5d398bf10.png)

@Mbonss

**Reply by Mbonss** [reply]
yes you are correct. Thank you for the quick reply!
